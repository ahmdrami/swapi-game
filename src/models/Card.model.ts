export type CardModel = {
  id: number
  name: string
  strength?: number
  color?: string
  url?: string
  winner?: boolean
}
