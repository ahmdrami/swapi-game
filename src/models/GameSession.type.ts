export type GameSessionModel = {
  result: string
  game: number
}
