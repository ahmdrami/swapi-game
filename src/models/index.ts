export * from './Card.model'
export * from './GameSession.type'
