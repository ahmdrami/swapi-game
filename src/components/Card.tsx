import { FunctionComponent } from 'react'
import { useHistory } from 'react-router-dom'
import { Flex, Heading } from 'theme-ui'

import { CardModel } from '../models'

export const Card: FunctionComponent<CardModel> = ({
  strength,
  name,
  url,
  color,
}): JSX.Element => {
  const history = useHistory()
  const handleClick = () => {
    if (url) {
      history.push(url)
    }
  }
  return (
    <Flex
      onClick={handleClick}
      sx={{
        height: 350,
        cursor: url ? 'pointer' : 'initial',
        pointerEvents: url ? 'auto' : 'none',
        bg: color,
        width: 250,
        p: 2,
        borderRadius: '5px',
        justifyContent: 'center',
        alignItems: 'center',
        flexDirection: 'column',
        '&:not(:last-child)': {
          mr: [0, 4],
          mt: [4, 0],
        },
        '> * ': {
          textAlign: 'center',
          color: 'black'
        },
      }}
    >
      <Heading as="h2" variant="styles.h2" sx={{ textTransform: 'capitalize' }}>
        {name}
      </Heading>
      {strength && (
        <Heading data-testid="strength" as="h3" variant="styles.h3">
          Strength: {strength}
        </Heading>
      )}
    </Flex>
  )
}
