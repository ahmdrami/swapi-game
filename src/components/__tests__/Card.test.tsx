import { render } from '@testing-library/react'
import { CardModel } from '../../models'
import { Card } from '../Card'

let mockCardData: CardModel
describe('Card component', () => {
  beforeEach(() => {
    mockCardData = {
      id: 1,
      name: 'Skywalker',
      color: '#111111',
      strength: 2,
    }
  })
  test('should render the fragment', () => {
    const { asFragment } = render(<Card {...mockCardData} />)

    expect(asFragment()).toMatchSnapshot()
  })

  test('should verify the title to exist in the document', () => {
    const { getByText, queryByTestId } = render(<Card {...mockCardData} />)

    expect(getByText(/Skywalker/)).toBeInTheDocument()
    expect(getByText(/Skywalker/).nodeName).toEqual('H2')
    expect(queryByTestId(/strength/)).toBeDefined()
  })

  test('should not display the strength when undefined in props', () => {
    delete mockCardData.strength
    const { queryByTestId } = render(<Card {...mockCardData} />)

    expect(queryByTestId(/strength/)).toBeNull()
  })
})
