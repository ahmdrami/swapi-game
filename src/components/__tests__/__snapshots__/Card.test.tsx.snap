// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should render the fragment 1`] = `
<DocumentFragment>
  <div
    class="css-1cslx62"
  >
    <h2
      class="css-19guh4r"
    >
      Skywalker
    </h2>
    <h3
      class="css-1wf7gym"
      data-testid="strength"
    >
      Strength: 2
    </h3>
  </div>
</DocumentFragment>
`;
